test/memkind_memtier_data_movement_test.cpp

```diff
@@ -1604,6 +1636,141 @@ TEST_F(RankingPerfTest, PageMovement)
+    int numa_id;
+
+    MTTInternalsLimits limits;
+    limits.lowLimit = TRACED_PAGESIZE;
+    limits.softLimit = 50ul * 1024ul * 1024ul;
+    limits.hardLimit = limits.softLimit;
+
+    mtt_allocator_create(&m_mtt_allocator, &limits);
+
+    // Allocate 2x time more data than soft limit so MTT will start moving data
+    // to PMEM
+    for (size_t i = 0; i < limits.softLimit * 2; i += alloc_size) {
+        void *ptr = mtt_allocator_malloc(&m_mtt_allocator, alloc_size);
+        memset(ptr, 1, alloc_size);
+        allocs.push_back(ptr);
+    }
+    std::cerr << "allocated " << allocs.size() << " objs" << std::endl;
+
+    // Wait for MTT to demote some pages to PMEM
+    sleep(5);
+
+    // Collect pages on DRAM and PMEM
+    for (auto it = allocs.begin(); it != allocs.end(); it++) {
+        get_mempolicy(&numa_id, nullptr, 0, (*it), MPOL_F_NODE | MPOL_F_ADDR);
+        if (dax_kmem_nodes.find(numa_id) != dax_kmem_nodes.end()) {
+            pmem_allocs.push_back((*it));
+        } else {
+            dram_allocs.push_back((*it));
+        }
+    }
+    std::cout << "dram_allocs.size() " << dram_allocs.size() << std::endl;
+    std::cout << "pmem_allocs.size() " << pmem_allocs.size() << std::endl;
+
+    // Check that that DRAM and PMEM contains similar number of pages
+    ASSERT_TRUE(dram_allocs.size() * 0.9 < pmem_allocs.size());
+    ASSERT_TRUE(pmem_allocs.size() * 0.9 < dram_allocs.size());
+
+    // Shuffle allocations to decrease caching effect
+    // NOTE that pages promotion/demotion is quite random
+    std::random_shuffle(dram_allocs.begin(), dram_allocs.end());
+    std::random_shuffle(pmem_allocs.begin(), pmem_allocs.end());
+
+    // We have to iterate over the same num of objects so choose min val to
+    // avoid overflows
+    const size_t num_objs = std::min(dram_allocs.size(), pmem_allocs.size());
+
+    // Wait for MTT to demote some pages to PMEM
+    std::cout << "sleep(5)" << std::endl;
+    sleep(5);
+
+    // Do some operations on DRAM memory, measure time
+    std::cout << "bench dram_allocs" << std::endl;
+    auto dram_time =
+        bench_vector(&dram_allocs, num_objs, alloc_size, iters_num);
+
+    // Do some operations on PMEM memory, measure time
+    std::cout << "bench pmem_allocs" << std::endl;
+    auto pmem_time1 =
+        bench_vector(&pmem_allocs, num_objs, alloc_size, iters_num);
+
+    ASSERT_TRUE(dram_time < pmem_time1);
+
+    // Wait for MTT to promote some pages to DRAM
+    std::cout << "sleep(5)" << std::endl;
+    sleep(5);
+
+    // Check, how many pages were promoted to DRAM
+    int num_moved = 0;
+    for (size_t it = 0; it < num_objs; it++) {
+        get_mempolicy(&numa_id, nullptr, 0, pmem_allocs[it],
+                      MPOL_F_NODE | MPOL_F_ADDR);
+        if (dram_nodes.find(numa_id) != dram_nodes.end())
+            num_moved++;
+    }
+    std::cout << "num_moved: " << num_moved << " "
+              << (float)num_moved / num_objs * 100.0 << "%" << std::endl;
```

Comment:
I think it would be best to surround the cast with brackets, or use ```static_cast<double>(num_moved);``` in order to avoid any possible confusion about cast ordering. Besides, why do we use single precision?